packages/react/src/Details/Details.docs.json

```diff
@@ -11,5 +11,26 @@
       "type": "SystemStyleObject"
     }
   ],
-  "subcomponents": []
+  "subcomponents": [
+    {
+      "name": "Details.Summary",
+      "props": [
+        {
+          "name": "as",
+          "type": "React.ElementType<React.PropsWithChildren<any>>",
+          "defaultValue": "summary",
+          "required": false,
+          "description": "HTML element to render summary as."
+        },
+        {
+          "name": "children",
+          "type": "React.ReactNode"
+        },
+        {
+          "name": "sx",
+          "type": "SystemStyleObject"
+        }
+      ]
+    }
+  ]
```

Comment:
(blocking) `summary` prop is missing from docs.

Curious, should we prefer adding just 1 approach instead of both `<Details summary>` and `<Details.Summary>`?

To me, `<Details.Summary>` feels like the closest replacement for `<summary>`, which feels good?

Reply:
Thought about this and I agree with you. I think originally I was going for the "maximize ease-of-use" approach but in hindsight this might be confusing and adds extra complexity with having to support the intersection of both strategies (Re: your comment [here](https://github.com/primer/react/pull/5015#discussion_r1784493723)).

Going to refactor to just introduce `Details.Summary` and render a default one if one isn't present, that feels right-er to me as well. 
Wondering, do you think that would warrant a warning as well? (The use case where a `Details` component is used but neither `Details.Summary` nor a native `summary` is present within the `children`).